plugin/pkg/admission/broker/authsarcheck/admission_test.go

```diff
@@ -0,0 +1,158 @@
+	kf := kubeinformers.NewSharedInformerFactory(kubeClient, 5*time.Minute)
+	handler, err := NewSARCheck()
+	if err != nil {
+		return nil, kf, err
+	}
+	pluginInitializer := scadmission.NewPluginInitializer(nil, nil, kubeClient, kf)
+	pluginInitializer.Initialize(handler)
+	err = admission.Validate(handler)
+	return handler, kf, err
+}
+
+// newMockKubeClientForTest creates a mock kubernetes client that is configured
+// to allow any SAR creations.
+func newMockKubeClientForTest() *kubefake.Clientset {
+	mockClient := &kubefake.Clientset{}
+	mockClient.AddReactor("create", "subjectaccessreviews", func(action core.Action) (bool, runtime.Object, error) {
+		mysar := &authorizationapi.SubjectAccessReview{
+			Status: authorizationapi.SubjectAccessReviewStatus{
+				Allowed: true,
+				Reason:  "seemed friendly enough",
+			},
+		}
+		return true, mysar, nil
+	})
+	return mockClient
+}
+
+// TestAdmissionBroker tests Admit to ensure that the result from the SAR check
+// is properly checked.
+func TestAdmissionBroker(t *testing.T) {
+	mockKubeClient := newMockKubeClientForTest()
+	handler, kubeInformerFactory, err := newHandlerForTest(mockKubeClient)
+	if err != nil {
+		t.Errorf("unexpected error initializing handler: %v", err)
+	}
+	kubeInformerFactory.Start(wait.NeverStop)
+
+	// Anonymous struct fields:
+	// name: short description of the testing
+	// broker: a fake broker object
+	// allowed: flag for whether or not the broker should be admitted
+	cases := []struct {
+		name    string
+		broker  *servicecatalog.ServiceBroker
+		allowed bool
+	}{
+		{
+			name: "broker with no auth",
+			broker: &servicecatalog.ServiceBroker{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-broker",
+				},
+				Spec: servicecatalog.ServiceBrokerSpec{
+					URL: "http://example.com",
+				},
+			},
+			allowed: false,
+		},
+		{
+			name: "broker with basic auth, same namespace",
+			broker: &servicecatalog.ServiceBroker{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-broker",
+					Namespace: "test-ns",
+				},
+				Spec: servicecatalog.ServiceBrokerSpec{
+					URL: "http://example.com",
+					AuthInfo: &servicecatalog.ServiceBrokerAuthInfo{
+						Basic: &servicecatalog.BasicAuthConfig{
+							SecretRef: &v1.ObjectReference{
+								Namespace: "test-ns",
+								Name:      "test-secret",
+							},
+						},
+					},
+				},
+			},
+			allowed: true,
+		},
+		{
+			name: "broker with bearer token, same namespace",
+			broker: &servicecatalog.ServiceBroker{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-broker",
+					Namespace: "test-ns",
+				},
+				Spec: servicecatalog.ServiceBrokerSpec{
+					URL: "http://example.com",
+					AuthInfo: &servicecatalog.ServiceBrokerAuthInfo{
+						Bearer: &servicecatalog.BearerTokenAuthConfig{
+							SecretRef: &v1.ObjectReference{
+								Namespace: "test-ns",
+								Name:      "test-secret",
+							},
+						},
+					},
+				},
+			},
+			allowed: true,
+		},
+	}
+
```

Comment:
we need a negative test where you pass an unexpected userinfo and get a negative result from the authorizer.